### PR TITLE
Start of global lint ruleset

### PR DESCRIPTION
**Pull request by hx-markterry**
A first iteration of this.

A few changes from the eslint standard.

Can be easily added to a project using `make-up`.

### CODE REVIEW

**Comment by michaelcarter** [comment]
+1

**Comment by hxoliverrumbelow** [comment]
:+1:

File: coding-principles.md [hunk 1]
```diff
@@ -85,4 +85,10 @@ Before placing a new system or end point live, the following points should be co
...
 
-The above would be performed in conjuction with IT and project owners would need to be mindful of the extras time this will incur before go live.
\ No newline at end of file
+The above would be performed in conjuction with IT and project owners would need to be mindful of the extras time this will incur before go live.
+
+## Linting:
+
+All new projects must have code linting enabled and configured to fail CI if the standard are not meet.
+
+The preferred way of implementing this is to see our [linting rules documentation](https://github.com/holidayextras/culture/javascript-linting-rules.md#adding-to-your-project), as this simplifies the approach provides further consistency among project.
```
Comment (pmcnr-hx) [inline coding-principles.md]: You can use a relative URL here: `[linting rules documentation](javascript-linting-rules.md#adding-to-your-project)`.

File: README.md [hunk 2]
```diff
@@ -16,6 +16,7 @@ This is a collection of documents that outline Holiday Extras' culture and devel
  * [Technical Planning Meetings](/technical-planning-meeting.md)
  * [Expedited Procedure](/expedited-procedure.md)
  * [CoffeeScript to JavaScript Guidelines](/coffeescript-to-javascript-guidelines.md)
+ * [Javascript linting Rules](/javascript-linting-rules.md)
```
Comment (pmcnr-hx) [inline README.md]: The `/` is redundant: `[Javascript linting Rules](javascript-linting-rules.md)` achieves the same and seems to be the way recommend in the GitHub documentation for relative links.

File: javascript-linting-rules.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
+# Javascript Linting Rules
+
+The following rules should be applied to all of our javascript projects.
+
+_A project may use additional rules to ones listed here, but these would not be part of the current standard and such rules are subject to change in the future._
+
+## Current Rules
+
+These are stored in the [.eslintrc](https://github.com/holidayextras/culture/.eslintrc) in this repository.
```
Comment (pmcnr-hx) [inline javascript-linting-rules.md]: Would it make more sense to link to the raw file here instead so it can be easily downloaded?

**Comment by pmcnr-hx** [comment]
:+1:

File: coding-principles.md [hunk 4]
```diff
@@ -85,4 +85,10 @@ Before placing a new system or end point live, the following points should be co
  1. Add new endpoint/system to PCI and penetration testing hosts list.
  1. Check results of next external security audit for potential issues in new system.
 
-The above would be performed in conjuction with IT and project owners would need to be mindful of the extras time this will incur before go live.
\ No newline at end of file
+The above would be performed in conjuction with IT and project owners would need to be mindful of the extras time this will incur before go live.
```
Comment (kevinhodges) [inline coding-principles.md]: should be `extra` time

File: coding-principles.md [hunk 5]
```diff
@@ -85,4 +85,10 @@ Before placing a new system or end point live, the following points should be co
...
+
+## Linting:
+
+All new projects must have code linting enabled and configured to fail CI if the standard are not meet.
```
Comment (kevinhodges) [inline coding-principles.md]: `standards are not met`

File: javascript-linting-rules.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+For an explanation of these rules please refer to the [ESLint documentation](http://eslint.org/docs/rules/).
+
+### TLDR
+
+For the lazy here is a brief description of the rules specified:
+
+1. `"quotes": 0`
+1. `"no-underscore-dangle": 0`  Allow variables beginning with `_`.
+1. `"curly": [2, "multi-line"]` Force curly braces except on single line statements.
+1. `"camelcase": [2, {"properties": "never"}]` Force camel case variables except on object properties.
```
Comment (kevinhodges) [inline javascript-linting-rules.md:20]: If now is the time to be discussing this, I don't think SB can adopt this rule from the off.  All of our work and current coding standards is use camelcase.  Can we strip this out for the launch?

File: javascript-linting-rules.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+1. `"space-in-parens": [2, "never"]` Don't allow spaces inside round brackets and content.
```
Comment (kevinhodges) [inline javascript-linting-rules.md:21]: IMO, 

`if( variable1 > variable2 ) {`

is more readable than

`if(variable1 > variable2) {`

File: javascript-linting-rules.md (hunk 6 above)
Comment (hxoliverrumbelow) [inline javascript-linting-rules.md:20]: This rule equates to "your variables are camelCase, but we'll allow snake_case names within object properties` eg:

```
var fooBar = {
  snake_case: true
};
```

is valid and this:

```
var voo_bar = {
  snake_case: true
};
```

is invalid

File: javascript-linting-rules.md (hunk 7 above)
Comment (hx-markterry) [inline javascript-linting-rules.md:21]: This is a subjective rule, but my reasoning for choosing it was that the majority of popular o/s javascript code uses the version without spaces, for example:

https://github.com/jashkenas/backbone/blob/master/backbone.js
https://github.com/jashkenas/underscore/blob/master/underscore.js
https://github.com/moment/moment/blob/develop/moment.js
https://github.com/substack/node-browserify/blob/master/index.js
https://github.com/facebook/react/blob/master/main.js

And other known styles:

https://github.com/airbnb/javascript
http://google-styleguide.googlecode.com/svn/trunk/javascriptguide.xml
https://github.com/feross/standard

File: javascript-linting-rules.md (hunk 6 above)
Comment (kevinhodges) [inline javascript-linting-rules.md:20]: Ah cool, I should've looked at the definition myself

+1

**Pull request merged**